test/integration/Elsa.Core.IntegrationTests/Scripting/JavaScript/JavaScriptExpressionsIntegrationTests.cs

```diff
@@ -25,6 +31,97 @@ public async Task RunningAWorkflowThatIncludesJsonStringifyAParsedObjectShouldNo
         }
 
+        [Fact(DisplayName = "JavaScript expressions cannot access configuration by default")]
+        public async Task JavaScriptCannotAccessConfigurationByDefault()
+        {
+            var services = new ServiceCollection()
+                .AddSingleton<IConfiguration>(_ => new ConfigurationBuilder()
+                    .AddInMemoryCollection(new Dictionary<string, string>
+                    {
+                        { "SomeSecret", "I am a secret" },
+                    })
+                    .Build())
+                .AddSingleton<AssertableActivityState>()
+                .AddElsa(elsa => elsa
+                    .AddActivity<AddExpressionToActivityState>()
+                    .AddWorkflow<ConfigurationAccessWorkflow>())
+                .BuildServiceProvider();
+
+            var workflowStarter = services.GetRequiredService<IBuildsAndStartsWorkflow>();
+            var activityState = services.GetRequiredService<AssertableActivityState>();
+
+            await workflowStarter.BuildAndStartWorkflowAsync<ConfigurationAccessWorkflow>();
+
+            Assert.Single(activityState.Messages, "");
+        }
+
+        [Fact(DisplayName = "JavaScript expressions can have configuration access enabled")]
+        public async Task JavaScriptExpressionsHaveConfigurationAccessEnabled()
+        {
+            var services = new ServiceCollection()
+                .AddSingleton<IConfiguration>(_ => new ConfigurationBuilder()
+                    .AddInMemoryCollection(new Dictionary<string, string>
+                    {
+                        { "SomeSecret", "I am a secret" },
+                    })
+                    .Build())
+                .AddSingleton<AssertableActivityState>()
+                .AddElsa(elsa => elsa
+                    .AddActivity<AddExpressionToActivityState>()
+                    .AddWorkflow<ConfigurationAccessWorkflow>())
+                .WithJavaScriptOptions(x =>
+                {
+                    x.EnableConfigurationAccess = true;
+                })
+                .BuildServiceProvider();
+
+            var workflowStarter = services.GetRequiredService<IBuildsAndStartsWorkflow>();
+            var activityState = services.GetRequiredService<AssertableActivityState>();
+
+            await workflowStarter.BuildAndStartWorkflowAsync<ConfigurationAccessWorkflow>();
+
+            Assert.Single(activityState.Messages, "I am a secret");
+        }
+
+        private class ConfigurationAccessWorkflow : IWorkflow
+        {
+            public void Build(IWorkflowBuilder builder)
+            {
+                builder.StartWith<AddExpressionToActivityState>(setup =>
+                {
+                    setup.Set(x => x.Expression, _ => "getConfig('SomeSecret')");
+                    setup.Set(x => x.ExpressionSyntax, _ => "JavaScript");
+                });
+            }
+        }
+
+        private class AddExpressionToActivityState : Activity
```

Comment:
I'm planning on adding this to the shared project later and using in the Liquid test. I'd also like to add an integration test for Clr access